site/docs/v1/tech/apis/recovery-codes.adoc

```diff
@@ -0,0 +1,93 @@
+---
+layout: doc
+title: Recovery Codes APIs
+description: APIs for creating and retrieving recovery codes for multi-factor authentication
+---
+
+== Overview
+Recovery codes can be used to complete a multi-factor login or disable multi-factor authentication. Typically they are used when a User has lost their device or retrieval access to all multi-factor methods.
+
+The following APIs are provided to manage Recovery Codes.
+
+* <<Create Recovery Codes>>
+* <<Retrieve Recovery Codes>>
+
+== Create Recovery Codes
+
+This API is used to create a list of Recovery Codes
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Create Recovery Codes
+[.endpoint]
+.URI
+--
+[method]#POST# [uri]#/api/recovery-codes/`\{userId\}`#
+--
+
+==== Request Parameters
+
+[.api]
+[field]#userId# [type]#[UUID]# [required]#Required#::
+The unique Id of the user to assign the generated Recovery Codes to.
+
+=== Response
+
+The response for this API contains an array of size 10 of Recovery Codes that were created.
+
+:never_search_error:
+include::docs/v1/tech/apis/_standard-post-response-codes.adoc[]
+:never_search_error!:
+
+==== Response Body
+
+[.api]
+[field]#recoveryCodes# [type]#[Array<String>]#::
+The array of Recovery Codes.
+
+[source,json]
+.Example Response JSON
+----
+include::docs/src/json/recovery-codes/response.json[]
+----
+
+== Retrieve Recovery Codes
+
+This API is used to retrieve Recovery Codes for a User
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Retrieve Recovery Codes
+[.endpoint]
+.URI
+--
+[method]#GET# [uri]#/api/recovery-codes/`\{userId\}`#
+--
+
+==== Request Parameters
+
+[.api]
+[field]#userId# [type]#[UUID]# [required]#Required#::
+The Id of the User to retrieve Recovery Codes for.
+
+=== Response
+
+The response for this API contains the Recovery Codes that are assigned to the User
```

Comment:
The response will contain the remaining recovery codes, each time one is used it is removed, so this response will contain between 0 and 10 codes.